signing/signer.go

```diff
@@ -114,22 +115,9 @@ func (es *EdSigner) PrivateKey() PrivateKey {
 }
 
 // VRFSigner wraps same ed25519 key to provide ecvrf.
-func (es *EdSigner) VRFSigner(opts ...VRFOptionFunc) (*VRFSigner, error) {
-	cfg := &vrfOption{}
-
-	for _, opt := range opts {
-		if err := opt(cfg); err != nil {
-			return nil, err
-		}
-	}
-
-	if err := cfg.validate(); err != nil {
-		return nil, err
-	}
-
+func (es *EdSigner) VRFSigner() (*VRFSigner, error) {
 	return &VRFSigner{
-		privateKey: es.priv,
+		privateKey: oasis.PrivateKey(es.priv),
```

Comment:
i am confused by this cast. why is this needed?

Reply:
`es.priv` is a `crypto.PrivateKey` defined in the standard library. The but the signer uses `"github.com/oasisprotocol/curve25519-voi/primitives/ed25519"` which for some reason defined its own type of `PrivateKey`. Both are just `[]byte` but there are no implicit casts in go (at least not from non-basic type to non-basic type), so this explicit cast is necessary.